client/firmwareOSource_service.go

```diff
@@ -80,7 +81,7 @@ func (sm *ServiceManager) UpdateOSource(name string, description string, firmwar
 	rn := fmt.Sprintf("osrc-%s", name)
 	parentDn := fmt.Sprintf("uni/fabric/fwrepop")
 	firmwareOSource := models.NewOSource(rn, parentDn, description, firmwareOSourceAttr)
-	firmwareOSource.Status = "modified"
+	firmwareOSource.Status = ""  // this avoids API failure in case MO is not created in APIC
```

Comment:
Why is this line necessary if NewOSource already returns a struct with Status set to empty string? If its not needed - remove it.

Reply:
Will take care of it.